alembic: Fix install destinations

hesiod opened the pull request:
~~Note: I've written a tester (`testImportedCMakePackage`) in order to prevent issues like the one described below in the future. It's written in a generic fashion and should be applicable to other CMake projects as well. It currently resides below alembic's directory, in `pkgs/development/libraries/alembic/cmakePackageTests`, as I'm not sure about the interest in such a tester. If there's interest in using it as a generic tester I could also move it to `pkgs/build-support/testers/`. It is somewhat comparable to the `hasPkgConfigModule` tester that is already located there.~~
Edit: I've moved the tester to a separate commit for now in order to streamline this PR a bit.

###### Description of changes
The generated CMake targets file was referring to an incorrect destination as the derivation manually moved the libraries during installPhase, while CMake uses the path it thinks is going to be used (the `DESTINATION` in the install rule) in the `IMPORTED_LOCATION` property.

By setting the install destinations via CMake flags (and patching the `DESTINATION` for the binary install rules), CMake will pick up the correct locations in the generated `AlembicTargets-release.cmake` file.

Along with fixing that issue, this commit also includes the following changes:
* Remove unused unzip nativeBuildInput
* Add missing direct dependency imath: Previously it was only picked up indirectly, resulting in CMake configuration warnings
* Add ~~imath~~ ilmbase as propagatedBuildInput: Downstream users of Alembic (via CMake) need to add ~~imath~~ ilmbase as a dependency as well. For some reason this is not discovered correctly otherwise
* Use CMake setup hooks instead of setting buildPhase/installPhase
* Add maintainer tmarkus

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hesiod committed:
blender: Fix build after changes in Alembic package

davidak commented:
Can you add the test (`testImportedCMakePackage`) in a separate commit? The file can later be moved if there is interest.

hesiod commented:
@davidak

> Can you add the test (testImportedCMakePackage) in a separate commit? The file can later be moved if there is interest.

Yeah, good point, I'll do that.

If you're reviewing this PR atm: Please note that there's a minor correction I've made locally I haven't pushed yet: `propagatedBuildOutputs` needs to contain `"lib"`. The full line should be `propagatedBuildOutputs = [ "lib" ];`.
Apparently I forgot to push that commit earlier. I won't force push right now so that the current check results remain visible.

davidak commented:
I tried to review and i haven't found an issue, but i'm not comfortable to review the cmake test, so i leave it to someone else.

Thanks for your efforts!

hesiod commented:
Marking this PR as a draft while I reorganize and improve it a bit

hesiod committed:
alembic: Fix install destinations

The generated CMake targets file was referring to an incorrect
destination as the derivation manually moved the libraries during
installPhase, while CMake uses the path it thinks is going to be used
(the DESTINATION in the install rule) in the IMPORTED_LOCATION property.

By setting the install destinations via CMake flags (and patching the
DESTINATION for the binary install rules), CMake will pick up the
correct locations in the generated AlembicTargets-release.cmake file.

Along with fixing that issue, this commit also includes the following
changes:
* Remove unused unzip nativeBuildInput
* Enable unit tests
* Add missing direct dependency ilmbase:
  Previously it was only picked up indirectly, resulting in CMake
  configuration warnings
* Add ilmbase as propagatedBuildInput:
  Downstream users of Alembic (via CMake) need to add ilmbase as a
  dependency as well
  For some reason this is not discovered correctly otherwise
* Use CMake setup hooks instead of setting buildPhase/installPhase

hesiod committed:
alembic: Add maintainer tmarkus

davidak commented on pkgs/development/libraries/alembic/default.nix:
```diff
@@ -12,26 +12,50 @@ stdenv.mkDerivation rec
-  '';
+  # NOTE: Alembic also support imath instead of ilmbase, but some users of Alembic (e.g. Blender)
+  # are incompatible with the imath version of Alembic
+  buildInputs = [ openexr hdf5-threadsafe ilmbase ];
```
is ilmbase needed here if it's already in `propagatedBuildInputs`? from my understanding is only one needed

davidak replied:
>* Add missing direct dependency ilmbase:
  Previously it was only picked up indirectly, resulting in CMake
  configuration warnings

i see

davidak commented:
Result of `nixpkgs-review pr 215528` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>blender</li>
    <li>blender-hip</li>
    <li>python310Packages.bpycv</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>alembic</li>
  </ul>
</details>

hesiod commented:
@davidak Thanks, I think those failures are fallout due to #212787 (more precisely, commit f386825fa23a7b1e67e0c93e82bcbd318da03c86).
See the following Hydra failures:
- embree: https://hydra.nixos.org/build/208695406
- openimagedenoise: https://hydra.nixos.org/build/208739835

I was going to create an issue, but it turns out issues already exist for those failures: #216580, #216390

davidak commented:
Result of `nixpkgs-review pr 215528` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>alembic</li>
    <li>alembic.bin</li>
    <li>alembic.dev</li>
    <li>alembic.lib</li>
    <li>blender</li>
    <li>blender-hip</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
  </ul>
</details>

davidak reviewed (approved):
Change looks good

Builds, works
